docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/How_Tos/Customise_React_Forms.md

```diff
@@ -0,0 +1,254 @@
+# Customising React Forms
+
+Forms that are rendered with React use the [ReduxForm](http://redux-form.com) library and are based on schema definitions that come from the server. To customise these forms, you can apply middleware that updates the schema or applies validation.
+
+## Toggling a form field
+
+Let's have a field hide or show based on the state of another field. We want the field "State" to show if `Country === 'US'`.
+
+First, we need to add a customisation to all form fields that allows them to be toggleable.
+
+_my-module/js/src/HideableComponent.js
+```js
+import React from 'react';
+
+const HideableComponent = ({Component, ...props}) => (
+  props.shouldHide ? null : <Component {...props} />
+);
+
+HideableComponent.propTypes = {
+  shouldHide: React.PropTypes.boolean
+};
+
+HideableComponent.defaultProps = {
+  shouldHide: false
+};
+
+export default (Component) => (props) => (
+  props.shouldHide ? null : <Component {...props} />
+);
+```
+
+Now, let's apply this through Injector.
+
+_my-module/js/main.js_
+```js
+Injector.transform(
+  'toggle-field',
+  (updater) => {
+    updater.react('ReduxFormField', HideableComponentCreator);
+  }
+);
+```
+
+Lastly, we need to apply a schema transformation using `updater.form.alterSchema()`.
+
+```js
+Injector.transform(
```

Comment:
Yeah, this is a mess. Very confusing for a new dev. I've been chatting to @flamerohr about how we can simplify this. The fundamental issue is that these form customisations are not middleware in the pure sense. They're more like reducers. I'm going to do a bit of investigation to see how we can get that callback signature to look more like you'd expect:

```js
(formValues, schemaUpdater) => schemaUpdater.getState();
```
without all of the `updateSchema` bloat.